### DISCUSSION TITLE
Converting pdf to typst::model::Document possible?

### DISCUSSION THREAD

**Discussion by KillTheMule** [typst/typst#5248] [General]
Reading https://github.com/typst/typst/issues/145 (and not wanting derail this issue further) I was wondering if it would be possible to read a pdf file and convert that to a typst `Document`. I know only a bit about `Document` and basically nothing about pdf's, and heuristically I'd assume the answer is 'no' (otherwise I'd have expected someone to bring that up in the issue), but if so, what would be the blocker(s)? Or would that "just" be a lot of work?

**Comment by laurmaedje** [comment]
A (rather lossy) conversion would be possible, but it basically means implementing a PDF viewer (sans rendering and interactive parts).

**Comment by KillTheMule** [comment]
What's the reason for it being "rather lossy"? Does pdf have elements that have no representation in a `Document`, or does the latter miss some flexibility/functionality? Or is that just related to the possibly interactive/animated parts of a pdf?

**Reply by laurmaedje** [reply]
PDF has a ton of features that Typst Documents do not support. Color space stuff, PostScript stuff, Annotations, etc.

**Comment by kierun** [comment]
There are PDF to text converters… I guess you could use that and “fix” things afterwards. Not ideal.
